bluefin-random/src/Bluefin/Random.hs

```diff
@@ -0,0 +1,45 @@
+module Bluefin.Random (
+  Random,
+  runRandom,
+  withRandom
+  ) where
+
+import Prelude ((.), pure, (<$>), ($), flip)
+
+import qualified System.Random as Rnd
+import Bluefin.State (State, evalState, get, put)
+import Bluefin.Eff (Eff, (:&))
+import qualified System.Random.Stateful as Rnd
+import Data.Bifunctor (second)
+
+newtype Random g e = Random (State g e)
+
+newtype RandomPure g = RandomPure { unRandomPure :: g }
+  deriving newtype (Rnd.RandomGen)
+
+instance Rnd.RandomGen g => Rnd.StatefulGen (Random g e) (Eff (e :& es)) where
+  uniformWord64 = flip withRandom Rnd.genWord64
+
+  uniformByteArrayM pinned size = flip withRandom
+    $ Rnd.uniformByteArray pinned size
+
+instance Rnd.RandomGen g => Rnd.FrozenGen (RandomPure g) (Eff (e :& es)) where
+  type MutableGen (RandomPure g) (Eff (e :& es)) = Random g e
+
+  modifyGen (Random s) f = withRandom (Random s)
+    $ second unRandomPure . f . RandomPure
+
+runRandom ::
+  g ->
+  (forall e. Random g e -> Eff (e :& es) a) ->
+  Eff es a
+runRandom g f = evalState g (f . Random)
+
+withRandom ::
+  Random g e ->
+  (g -> (a, g)) ->
+  Eff (e :& es) a
```

Comment:
This looks like `applyRandomGenM` from class `RandomGenM`, so maybe we should make it so?

Reply:
Ah, so it is. I was looking at old docs.

Reply:
Looks like this has been enacted.